modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/to/Chapters.kt

```diff
@@ -1,6 +1,12 @@
 package au.com.shiftyjelly.pocketcasts.models.to
 
-data class Chapters(private val items: List<Chapter> = emptyList()) {
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.Feature
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.FeatureFlagWrapper
+
+data class Chapters(
+    private val items: List<Chapter> = emptyList(),
+    private val featureFlagWrapper: FeatureFlagWrapper = FeatureFlagWrapper(),
```

Comment:
It feels weird to have a `data class` contain something like a `FeatureFlagWrapper`. I'm not sure why we need this class. Feature flag values can be mocked in tests with providers like here.

https://github.com/Automattic/pocket-casts-android/blob/c502a6267b58b305cd708d955e5f977976ae9986/app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/repositories/sync/PodcastSyncProcessTest.kt#L65-L73

I might not have enough context but if this is something that we do often we could just create a provider like this and reuse it in tests.

```kotlin
class InMemoryFeatureProvider : ModifiableFeatureProvider {
    private val features = mutableMapOf<Feature, Boolean>()

    override val priority = 0

    override fun setEnabled(feature: Feature, enabled: Boolean) {
        features[feature] = enabled
    }

    override fun hasFeature(feature: Feature) = true

    override fun isEnabled(feature: Feature) = features[feature] ?: feature.defaultValue
}
```

Possibly even as a rule that initializes and clears providers so we don't need to register it every time.

Reply:
Love the idea! I applied it at https://github.com/Automattic/pocket-casts-android/pull/1874/commits/eac5ca1e8f37370859cc6aa058de7a02435ab15a.

I'll check about creating a rule and address it in another PR.